tests_oval_graph/test_tools.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+import re
+import tempfile
+import uuid
+from pathlib import Path
+
+import pytest
+
+TOP_PATH = Path(__file__).parent
+
+
+class TestTools():
+    @staticmethod
+    def find_files(file_name, search_path):
+        result = []
+        # root, directory, files
+        for root, _, files in os.walk(search_path):
+            for filename in files:
+                if file_name in filename:
+                    result.append(os.path.abspath(os.path.join(root, filename)))
+        return result
+
+    @staticmethod
+    def get_text_file(src):
+        path = TOP_PATH / src
+        with open(path, 'r') as data:
+            return data.readlines()
+
+    @staticmethod
+    def get_random_dir_in_tmp():
+        return os.path.join(tempfile.gettempdir(), str(uuid.uuid4()))
+
+    @staticmethod
+    def compare_results_html(result):
+        result_ = TestTools.get_text_file(result)
+        reference_pattern = TestTools.get_text_file(
+            'test_commands/test_data/referenc_pattern_html_report.txt')
+        prefix_start = '<script>var data_of_tree = '
+        prefix_end = ';</script><div>\n'
+        data_in_html = ""
+        matched = False
+        for row in result_:
+            if prefix_start in row and prefix_end in row:
+                matched = True
+                data_in_html = row
+        assert matched
+
+        tmp_json_str = data_in_html.replace(prefix_start, '').replace(prefix_end, '')
+        tmp_json = json.loads(tmp_json_str)
+        data_in_html = prefix_start + json.dumps(tmp_json, indent=4, sort_keys=False) + prefix_end
+
+        count_row = 0
+        rule_name = 'xccdforgssgprojectcontentrulepackageabrtremoved'
```

Comment:
What is this?

Reply:
It is the ID of the rule in the HTML report. Because jQuery has a problem with non-alphabet characters.